charts/ota-mariadb/templates/stateful-set.yaml

```diff
@@ -0,0 +1,71 @@
+---
+apiVersion: apps/v1beta2
+kind: StatefulSet
+metadata:
+  name: {{ template "ota-mariadb.fullname" . }}
+  labels:
+    app: {{ template "ota-mariadb.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  serviceName: mysql
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ template "ota-mariadb.fullname" . }}
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "ota-mariadb.fullname" . }}
+        release: {{ .Release.Name }}
+    spec:
+      containers:
+      - image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        name: {{ template "ota-mariadb.fullname" . }}
+        envFrom:
+        - secretRef:
+            name: {{ template "ota-mariadb.fullname" . }}
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: {{ .Values.service.internalPort }}
+          name: {{ .Values.service.name }}
+        livenessProbe:
+          exec:
+            command: ["mysqladmin", "ping"]
+          initialDelaySeconds: 30
+          periodSeconds: 10
+          timeoutSeconds: 5
+        readinessProbe:
+          exec:
+            # Check we can execute queries over TCP (skip-networking is off).
+            command: ["mysql", "-h", "127.0.0.1", "-proot", "-e", "SELECT 1"]
+          initialDelaySeconds: 5
+          periodSeconds: 2
+          timeoutSeconds: 1
+        volumeMounts:
+        - name: {{ template "ota-mariadb.fullname" . }}-files
+          mountPath: /etc/mysql/mariadb.conf.d/
+        - name: mysql-data
+          mountPath: /var/lib/mysql
+    {{- with .Values.nodeSelector }}
+      nodeSelector:
+{{ toYaml . | indent 8 }}
+    {{- end }}
+      volumes:
+      - name: {{ template "ota-mariadb.fullname" . }}-files
+        configMap:
+          name: {{ template "ota-mariadb.fullname" . }}-files
+      - name: mysql-data
+      {{- if .Values.persistence.enabled }}
+        persistentVolumeClaim:
```

Comment:
Don't you need _volumeClaimTemplates_ when in a statefulset? I assume this was tested so maybe I am misled here.